internal/controller/deployment_to_pdb_controller.go

```diff
@@ -233,30 +208,45 @@ func triggerOnAnnotationChange(e event.UpdateEvent, logger logr.Logger) bool {
 	return false
 }
 
-// hasNonZeroMaxUnavailable returns true if the deployment has maxUnavailable set to a non-zero value.
-// Deployments with maxUnavailable != 0 already tolerate downtime, so PDB creation is skipped.
-func hasNonZeroMaxUnavailable(deployment *v1.Deployment) bool {
-	if deployment.Spec.Strategy.RollingUpdate == nil {
-		return false
-	}
-	maxUnavailable := deployment.Spec.Strategy.RollingUpdate.MaxUnavailable
-	if maxUnavailable == nil {
-		return false
-	}
-	if maxUnavailable.Type == intstr.Int {
-		return maxUnavailable.IntVal != 0
-	}
-	// String type - check for "0" or "0%"
-	return maxUnavailable.StrVal != "0" && maxUnavailable.StrVal != "0%"
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *DeploymentToPDBReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	logger := mgr.GetLogger()
 	// Set up the controller to watch Deployments and trigger the reconcile function
 	// when controller restarts everything is seen as a create event
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1.Deployment{}).
+		Watches(&corev1.Namespace{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
```

Comment:
this is  a little unconventional for a controller so need a big comment here but anytime a nameapace is modified you're checking if it has an annotation or is in the hardcoded list then re-enqueing all deploments there in to recocnile. 

That helps enabling but not disabling. And you do have an oddity where if soemthign is constatnly updating a namepace you may create ALOT of churn. (you don't have a way of seeing if the namespace transitioned just htat its on.